src/client/Client.cc

```diff
@@ -2539,6 +2539,31 @@ void Client::handle_client_request_forward(const MConstRef<MClientRequestForward
   MetaRequest *request = mds_requests[tid];
   ceph_assert(request);
 
+  auto num_fwd = fwd->get_num_fwd();
+  if (num_fwd <= request->num_fwd) {
+    /*
+     * The type of 'num_fwd' in ceph 'MClientRequestForward'
+     * is 'int32_t', while in 'ceph_mds_request_head' the
+     * type is '__u8'. So in case the request bounces between
+     * MDSes exceeding 256 times, the client will get stuck.
+     *
+     * In this case it's ususally a bug in MDS and continue
+     * bouncing the request makes no sense.
+     */
+    if (request->num_fwd == 256) {
```

Comment:
How about `request->num_fwd > UCHAR_MAX` ? Or even better, express it in terms of sizeof() the field in the ceph_mds_request_head.

Reply:
Done.